builder/penv_setup.py

```
@@ -0,0 +1,348 @@
+
+    try:
+        subprocess.check_call([
+            uv_executable, "pip", "install", "--quiet", "--force-reinstall",
+            f"--python={python_exe}",
+            "-e", esptool_repo_path
+        ])
+
+    except subprocess.CalledProcessError:
+        sys.exit(1)
+
+    except subprocess.CalledProcessError as e:
```

Ruff (0.12.2)

try-except block with duplicate exception `subprocess.CalledProcessError`

(B025)